src/routes/relay/relay.controller.spec.ts

```diff
@@ -0,0 +1,1646 @@
+            .with('value', faker.number.bigInt())
+            .encode() as Hex;
+          const taskId = faker.string.uuid();
+          networkService.get.mockImplementation((url) => {
+            switch (url) {
+              case `${safeConfigUrl}/api/v1/chains/${chainId}`:
+                return Promise.resolve({ data: chain, status: 200 });
+              case `${chain.transactionService}/api/v1/safes/${safeAddress}`:
+                // Official mastercopy
+                return Promise.resolve({ data: safe, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+          networkService.post.mockImplementation((url) => {
+            switch (url) {
+              case `${relayUrl}/relays/v2/sponsored-call`:
+                return Promise.resolve({ data: { taskId }, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+
+          await request(app.getHttpServer())
+            .post(`/v1/chains/${chain.chainId}/relay`)
+            .send({
+              to: safeAddress,
+              data,
+            })
+            .expect(201)
+            .expect({
+              taskId,
+            });
+        });
+
+        it('should return 201 with manual gasLimit', async () => {
+          const chainId = faker.helpers.arrayElement(supportedChainIds);
+          const chain = chainBuilder().with('chainId', chainId).build();
+          const safe = safeBuilder().build();
+          const safeAddress = getAddress(safe.address);
+          const gasLimit = faker.number.bigInt();
+          const data = execTransactionEncoder().encode() as Hex;
+          const taskId = faker.string.uuid();
+          networkService.get.mockImplementation((url) => {
+            switch (url) {
+              case `${safeConfigUrl}/api/v1/chains/${chainId}`:
+                return Promise.resolve({ data: chain, status: 200 });
+              case `${chain.transactionService}/api/v1/safes/${safeAddress}`:
+                // Official mastercopy
+                return Promise.resolve({ data: safe, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+          networkService.post.mockImplementation((url) => {
+            switch (url) {
+              case `${relayUrl}/relays/v2/sponsored-call`:
+                return Promise.resolve({ data: { taskId }, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+
+          await request(app.getHttpServer())
+            .post(`/v1/chains/${chain.chainId}/relay`)
+            .send({
+              to: safeAddress,
+              data,
+              gasLimit: gasLimit.toString(),
+            })
+            .expect(201)
+            .expect({
+              taskId,
+            });
+
+          // The gasLimit should have a buffer added
```

Comment:
Not strictly related to this PR, but if I understand correctly, when the user specifies a custom `gasLimit`, 150k wei is added to the value the user specified. Do you think we should document this publicly in our API (Swagger maybe)?

Reply:
I'm not sure about the importance of this as the Gateway is inherently for internal use. Nonetheless, I've added a TODO to add it in a followup PR. There is a comment regarding the buffer in the `GelatoApi` as well.